### PR TITLE
fix real/imag color binding

### PR DESCRIPTION
**Pull request by jmfranck**

### CODE REVIEW

**Comment by wfkincaid** [comment]
This looks good and flake is already happy

File: pyspecdata/plot_funcs/pcolormesh.py [hunk 1]
```diff
@@ -54,7 +53,7 @@ def pcolormesh(
             fig = plt.gcf()
             ax1 = fig.add_subplot(1, 1, 1)
         ax_list = [(ax1, lambda x: x.real, "real")]
-    X, Y = np.meshgrid(
+    X, Y = plt.meshgrid(
```
Comment (jmfranck) [inline pyspecdata/plot_funcs/pcolormesh.py:56]: I might have said incorrectly before, but this ultimately comes from numpy -- so not changing this is better.

File: pyspecdata/core.py [hunk 2]
```diff
@@ -578,9 +578,10 @@ def lookup_rec(A, B, indexpair):
             )
         )
     whichisindex = joined[0][1].dtype.names.index(indexpair[1])
-    allbutindex = (
-        lambda x: list(x)[0:whichisindex] + list(x)[whichisindex + 1 :]
-    )
+
+    def allbutindex(x):
+        return list(x)[0:whichisindex] + list(x)[whichisindex + 1 :]
+
```
Comment (jmfranck) [inline pyspecdata/core.py]: it seemed that there was nothing in core.py that we would want, so I just reverted to the version that's currently in core (after flake8 compliance)

**Review by jmfranck — COMMENTED** [review]
again, make sure this works for you -- I also gave figlist the same treatment as core

**Pull request merged**